master/buildbot/test/unit/test_mybuilds.py

```diff
@@ -41,8 +27,97 @@ def test_prepare_display_repositories(self):
+            'branch': 'master',
+            'codebase': 'fmod',
+            'repository': 'https://github.com/example/repo.git',
+            'revision': 'bbbbbbbbbbbbaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'short_revision': 'bbbbbbbbbbbb',
+            'sourcestampsetid': 69
+        }]
+        status = self.get_mocked_status()
+        status.get_rev_url =\
+            lambda rev, repo: "https://github.com/example/repo/commit/bbbbbbbbbbbbaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+
+        expected_data = [{
+            'buildername': u'Elementary Test',
+            'builds_id': 131,
+            'builds_number': 4,
+            'complete': True,
+            'complete_at': '2018-03-22 13:07:17+00:00',
+            'project': 'Developer Tests',
+            'query_params': ['fmod_branch=master'],
+            'reason': "A build was forced by 'pyflakes pyflakes@localhost': ",
+            'slavename': 'build-slave-1',
+            'sourcestamps': [{'branch': 'master',
+                              'codebase': 'fmod',
+                              'display_repository': 'https://github.com/example/repo.git',
+                              'repository': 'https://github.com/example/repo.git',
+                              'revision': 'bbbbbbbbbbbbaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+                              'revision_url': 'https://github.com/example/repo/commit/bbbbbbbbbbbbaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+                              'short_revision': 'bbbbbbbbbbbb',
+                              'sourcestampsetid': 69}],
+            'sourcestampsetid': 69,
+            'submitted_at': '2018-03-22 13:07:11+00:00'
+        }]
 
+        builds = self.mybuilds.merge_sourcestamps_to_build(builds_by_ssid, display_repositories,
+                                                           sourcestamps, status)
+
+        self.assertEqual(builds, expected_data)
+
+
+    def test_prepare_builds_by_ssid(self):
+        builds =[
+            {'sourcestampsetid': 1, 'brid': 11,
+             'submitted_at': datetime.datetime(2018, 1, 1),
+             'complete_at': datetime.datetime(2018, 1, 2),
+            },
+            {'sourcestampsetid': 2, 'brid': 22,
+             'submitted_at': datetime.datetime(2018, 1, 3),
+             'complete_at': datetime.datetime(2018, 1, 4),
+            },
+        ]
+        expected_builds = {
+            1: {'sourcestampsetid': 1, 'brid': 11, 'sourcestamps': [], 'query_params': [],
+                'submitted_at': str(datetime.datetime(2018, 1, 1)),
+                'complete_at': str(datetime.datetime(2018, 1, 2))},
+            2: {'sourcestampsetid': 2, 'brid': 22, 'sourcestamps': [], 'query_params': [],
+                'submitted_at': str(datetime.datetime(2018, 1, 3)),
+                'complete_at': str(datetime.datetime(2018, 1, 4))},
+        }
+
+        builds_by_ssid = self.mybuilds.prepare_builds_by_ssid(builds)
+
+        self.assertEqual(sorted(builds_by_ssid.keys()), [1, 2])
```

Comment:
minor: this assertion doesn't check anything. In this way, you can do it easier:

```
for key in builds_by_ssid.keys():
            self.assertEqual(builds_by_ssid[key], expected_builds[key])
```

Reply:
Done